### ISSUE TITLE
Uncaught TypeError: Cannot read property 'callback' of undefined

### ISSUE THREAD

**Issue by hedefalk** [atom/atom#10432]
I guess this is my own code causing this, but can't seem to figure it out.

[Enter steps to reproduce below:]

1. install package ensime-atom (java and sbt and an sbt project and the whole shebang unfortunately)
2. start an ensime
3. stop it
4. start it again
5. stop it - now it blows up:

The code here:

https://github.com/atom/atom/blob/master/src/command-registry.coffee#L251

compiles to:
```
for (_j = 0, _len1 = listeners.length; _j < _len1; _j++) {
          listener = listeners[_j];
          if (immediatePropagationStopped) {
            break;
          }
          listener.callback.call(currentTarget, dispatchedEvent);
        }
``` 

and for some reason `listener = listeners[_j];`turns out to be undefined. Don't ask me why if this is single-threaded :)

**Atom Version**: 1.4.0-beta3
**System**: Mac OS X 10.11.2
**Thrown From**: Atom Core

### Stack Trace

Uncaught TypeError: Cannot read property 'callback' of undefined

```
At /Applications/Atom Beta.app/Contents/Resources/app.asar/src/command-registry.js:260

TypeError: Cannot read property 'callback' of undefined
    at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (/Applications/Atom Beta.app/Contents/Resources/app.asar/src/command-registry.js:260:19)
    at __bind (/Applications/Atom Beta.app/Contents/Resources/app.asar/src/command-registry.js:3:61)
    at KeymapManager.module.exports.KeymapManager.dispatchCommandEvent (/Applications/Atom Beta.app/Contents/Resources/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:536:16)
    at KeymapManager.module.exports.KeymapManager.handleKeyboardEvent (/Applications/Atom Beta.app/Contents/Resources/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:359:22)
    at WindowEventHandler.module.exports.WindowEventHandler.handleDocumentKeydown (/Applications/Atom Beta.app/Contents/Resources/app.asar/src/window-event-handler.js:97:36)
    at HTMLDocument.__bind (/Applications/Atom Beta.app/Contents/Resources/app.asar/src/window-event-handler.js:3:61)
```

### Commands

```
     -2:58.5.0 core:confirm (div#select-file.select-list.fuzzy-finder)
     -2:28.8.0 command-palette:toggle (atom-text-editor.editor.is-focused.hyperclick)
     -2:25.5.0 core:move-down (atom-text-editor.editor.mini.is-focused)
     -2:25.2.0 core:confirm (atom-text-editor.editor.mini.is-focused)
     -2:25.2.0 ensime:start (atom-text-editor.editor.hyperclick)
     -2:24 core:confirm (div#select-file.select-list.fuzzy-finder)
     -2:19.3.0 command-palette:toggle (atom-workspace.workspace.scrollbars-visible-when-scrolling.theme-base16-tomorrow-dark-theme.theme-atom-dark-ui)
     -2:17.2.0 core:confirm (atom-text-editor.editor.mini.is-focused)
     -2:17.2.0 ensime:stop (atom-workspace.workspace.scrollbars-visible-when-scrolling.theme-base16-tomorrow-dark-theme.theme-atom-dark-ui)
     -2:15.2.0 core:confirm (div#select-file.select-list.fuzzy-finder)
     -1:01.6.0 editor:consolidate-selections (atom-text-editor.editor.is-focused)
     -1:01.6.0 core:cancel (atom-text-editor.editor.is-focused)
     -1:00.3.0 command-palette:toggle (atom-text-editor.editor.is-focused.hyperclick)
     -0:58.5.0 core:confirm (atom-text-editor.editor.mini.is-focused)
     -0:58.5.0 ensime:stop (atom-text-editor.editor.hyperclick)
     -0:57.9.0 core:confirm (div#select-file.select-list.fuzzy-finder)
```

### Config

```json
{
  "core": {
    "ignoredNames": [
      ".git",
      ".ensime_cache",
      ".settings",
      ".idea",
      ".target",
      ".cache-main",
      ".cache-tests",
      "target"
    ],
    "packagesWithKeymapsDisabled": [
      "emmet"
    ],
    "themes": [
      "atom-dark-ui",
      "base16-tomorrow-dark-theme"
    ]
  },
  "editor": {
    "fontFamily": "menlo",
    "fontSize": 13,
    "invisibles": {},
    "preferredLineLength": 120,
    "showInvisibles": true,
    "softWrap": true
  }
}
```

### Installed Packages

```coffee
# User
Ensime, v0.31.1
Sublime-Style-Column-Selection, v1.4.0
atomatigit, v1.5.5
auto-detect-indentation, v0.4.2
auto-indent, v0.5.0
autocomplete-modules, v1.3.1
emmet, v2.3.17
hyperclick, v0.0.35
indentation-indicator, v0.6.0
js-hyperclick, v1.4.1
language-diff, v0.4.0
language-scala, v1.1.4
language-vue, v0.8.0
language-vue-component, v0.4.4
last-cursor-position, v0.9.0
latex, v0.28.2
linter, v1.11.3
linter-coffeelint, v1.1.2
linter-jshint, v1.2.2
minimap, v4.19.0
pretty-json, v0.4.1
project-manager, v2.7.3

# Dev
No dev packages
```

**Comment by hedefalk** [comment]
Listeners seems to have been removed:

<img width="1027" alt="screen shot 2016-01-15 at 15 48 56" src="https://cloud.githubusercontent.com/assets/116520/12355831/a62c505e-bb9f-11e5-89b2-c1c4cf8ecde3.png">

**Comment by hedefalk** [comment]
This seems to be a combination of this loop here in atom core: https://github.com/atom/atom/blob/master/src/command-registry.coffee#L251

and my own pretty half-arsed usage of vue.js here:

https://github.com/hedefalk/ensime-atom/blob/master/lib/views/select-file.coffee#L24

@yyx990803, do you have any idea what I'm doing wrong? It seems to me that the compiled coffescript code in atom's command registry (the one you see on my screen dump) breaks because I somehow remove the listeners from the list it is iterating on in one of the callbacks after it has decided the size of the array? Pretty nasty leakage that I don't understand.

**Comment by hedefalk** [comment]
Actually, the problem went away after registering my commands on my inner element instead of "window". Still weird though and I don't understand shit. I could also get rid of this error if I used stopImmediatePropagation instead of stopPropagation but then my own stuff didn't work. I actually called panel.destroy() but it was still visible.

Anyways, I think I just gotta let this one go since nobody else seems interested. I probably just did a general WTF. It's hard though, not understanding. Crap.

I'll close.
